lib/ansible/runner.rb

```diff
@@ -170,26 +175,36 @@ def async?(ansible_runner_method)
         ansible_runner_method == "start"
       end
 
-      def runner_params(base_dir, ansible_runner_method, extra_vars, playbook_or_role_args)
+      def runner_params(base_dir, ansible_runner_method, extra_vars, tags, playbook_or_role_args)
         runner_args = playbook_or_role_args.dup
 
         runner_args.delete(:roles_path) if runner_args[:roles_path].nil?
         skip_facts = runner_args.delete(:role_skip_facts)
         runner_args[:role_skip_facts] = nil if skip_facts
 
+        cmdline_commands = set_cmdline_commands(extra_vars, tags)
+
         runner_args[:ident]   = "result"
         runner_args[:hosts]   = "localhost"
-        runner_args[:cmdline] = AwesomeSpawn.build_command_line(nil, [{:extra_vars => extra_vars.to_json}]).lstrip if extra_vars.any?
+        runner_args[:cmdline] = AwesomeSpawn.build_command_line(nil, [cmdline_commands]).lstrip if extra_vars.any?
 
         [ansible_runner_method, base_dir, :json, runner_args]
       end
 
+      def set_cmdline_commands(extra_vars, tags)
+        commands = {:extra_vars => extra_vars.to_json}
+        commands[:tags] = tags if tags.present?
+
+        commands
+      end
+
       # Asserts passed parameters are correct, if not throws an exception.
-      def validate_params!(env_vars, extra_vars, ansible_runner_method, playbook_or_role_args)
+      def validate_params!(env_vars, extra_vars, tags, ansible_runner_method, playbook_or_role_args)
         errors = []
 
         errors << "env_vars must be a Hash, got: #{hash.class}" unless env_vars.kind_of?(Hash)
         errors << "extra_vars must be a Hash, got: #{hash.class}" unless extra_vars.kind_of?(Hash)
+        errors << "tags must be a String, got: #{String}" if tags.present? && !tags.kind_of?(String)
```

Comment:
> got: #{String}"

Pretty sure you mean `got: #{tag.class}` here

Reply:
@agrare Oh, sorry for that, changed it here for `tag.class`.